scripts/edit_media_directories.py

```diff
@@ -20,25 +31,43 @@ def get_usb_drives(logger: logging.Logger):
 
 def edit_directories(logger: logging.Logger):
   """Update LMS server with list of drives"""
-  usb_drives = get_usb_drives(logger) # Get list of available drives
-  usb_drives.append("") # Add blank drive to reflect empty string that is always at the end of the mediadirs section of request body
-  url = 'http://localhost:9000/settings/server/basic.html'
-  data={
-    'saveSettings': '1',
-    'useAJAX': '0',
-    'page': 'BASIC_SERVER_SETTINGS',
-    'pref_language': 'EN',
-    'pref_libraryname': '',
-    'pref_playlistdir': '',
-    'pref_rescantype': '1rescan'
-  } # All default headers, mediadirs added in loop
-  for d, drive in enumerate(usb_drives):
-    data[f"pref_mediadirs{d}"] = drive
-    data[f"pref_ignoreInAudioScan{d}"] = 1
-
-  logger.info("Adding drives to LMS settings...")
-  requests.post(url, data=data, verify=False, timeout=5)
-  logger.info("Drives added, LMS scanning now.")
+  try:
+    usb_drives = get_usb_drives(logger) # Get list of available drives
+    LMSMode = CheckLMSMode()
+    if LMSMode:
+      usb_drives.append("") # Add blank drive to reflect empty string that is always at the end of the mediadirs section of request body
+      url = 'http://localhost:9000/settings/server/basic.html'
+      data={
+        'saveSettings': '1',
+        'useAJAX': '0',
+        'page': 'BASIC_SERVER_SETTINGS',
+        'pref_language': 'EN',
+        'pref_libraryname': '',
+        'pref_playlistdir': '',
+        'pref_rescantype': '1rescan'
+      } # All default headers, mediadirs added in loop
+      for d, drive in enumerate(usb_drives):
+        data[f"pref_mediadirs{d}"] = drive
+        data[f"pref_ignoreInAudioScan{d}"] = 1
+
+      logger.info("Adding drives to LMS settings...")
+      requests.post(url, data=data, verify=False, timeout=5)
+      logger.info("Drives added, LMS scanning now.")
+    else:
+      if len(usb_drives) > 0:
+        drives = ""
+        for drive in usb_drives:
+          drives += f"{drive}, "
+        logger.info(f"Detected drives: {drives}")
+        logger.info("If you wish to mount these to the internal LMS server, please go to Config and set LMS Mode")
+  except (urllib3.exceptions.MaxRetryError, urllib3.exceptions.NewConnectionError, requests.exceptions.ConnectionError):
+    # This error seemingly only occurs on startup, leading me to believe it is a race condition
```

Comment:
Have we made any attempt to delay this service? Maybe make it dependent on the Logitech media server?

Reply:
I've made a few attempts in my last few commits, between creating the function that checked if lms was running to the systemd dependency attempts, it gives the same results of having this error print out 4 times during startup:
>Jun 25 15:30:10 lamplipi udisks2-listener.sh[551]: LMS Automated Drive Mounting has encountered an error: HTTPConnectionPool(host='localhost', port=9000): Max retries exceeded with url: /settings/server/basic.html (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x75db4390>: Failed to establish a new connection: [Errno 111] Connection refused'))

If there's another way to do this that I'm not aware of, please let me know